### PR TITLE
drop python 3.6 from bk

### PR DESCRIPTION
**Pull request by yuhan**

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9173" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9173?utm_source=stack-comment).

**Comment by alangenfeld** [comment]
we should correct all the `setup.py` of the packages and drop `3.6` and add `3.10` 
https://github.com/dagster-io/dagster/blame/master/python_modules/dagster/setup.py#L40-L43

File: .buildkite/dagster-buildkite/setup.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
     description="Tools for buildkite automation",
     url="https://github.com/dagster-io/dagster/tree/master/.buildkite/dagster-buildkite",
     classifiers=[
-        "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
```
Comment (dpeng817) [inline .buildkite/dagster-buildkite/setup.py:14]: should we add 3.10 here?

**Review by dpeng817 — APPROVED** [review]
Other than the one comment about 3.10, approving to unblock. One question; is it wise to start espousing support for 3.10 when we don't have it running against master atm? Pretty sure some of our type annotations features don't work on 3.10 currently, for example. cc @alangenfeld

**Comment by alangenfeld** [comment]
we do test master against 3.10, the deferred type hint thing got pushed back out of 3.10 

> We’re delaying making PEP 563 string-based annotations the default until Python 3.11

https://mail.python.org/archives/list/python-dev@python.org/thread/CLVXXPQ2T2LQ5MP2Y53VVQFCXYWQJHKZ/

**Review by alangenfeld — APPROVED** [review]

**Comment by chrish42** [comment]
FYI, there are still references to Python 3.6 in the `pyproject.toml` file. Also with the minimum version being 3.7, the [pep562](https://pypi.org/project/pep562/) dependency isn't needed anymore.

**Pull request merged**